src/main/java/org/opensearch/search/relevance/transformer/kendraintelligentranking/client/KendraHttpClient.java

```diff
@@ -34,6 +34,7 @@
 import java.security.AccessController;
```

Comment:
> I guess `setting` class is more specific towards that transformer itself, whereas the `configuration` is the schema of the search workflow. There are two things I am not sure if we should put in this PR:
> 
> 1. Can I use multiple same reranker, with different setting and configuration?
> 2. Do we want to provide more flexibility, such as one transformer depend on several other transformers.

1. In the long run, that's the intention. I think it's out of the scope of this specific refactoring.
2. Transformers are not really intended to depend on each other. They get chained so one ends up impacting what the next one sees. The assumption, though, is that a transformer should be unaware of its neighbors.

Reply:
Regarding the AccessController, this is a problem with the plugin model that we inherited, since it relies heavily on SecurityManager, which also marked for deprecation. 

AFAIK, the fix will come with the broader OpenSearch move to extensions, which will use a more robust isolation model (while things that do provide core functionality will be moved into core).